Packages/blueberry-jam-core/Runtime/LevelLoader/SceneTransitionerHelper.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections;
+using Codice.Client.BaseCommands.TubeClient;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BJ
+{
+    internal class SceneTransitionerHelper : SingletonGameObject<SceneTransitionerHelper>
```

Comment:
I noticed that for "CoroutineHelper.cs" you defined the CoroutineHelper class as internal though for the pause manager you had no internal classes.

As far as I understand, an internal class can only be accessed by other classes within the same namespace (in this case BJ namespace) and it can only be accessed within the Blueberry Jam Core package. Please correct me if I am wrong.

I am curious to know the reason why you did this? Is it to provide a short, easy-to-understand list of functionality to the end user (this is the reason I did it for the LevelLoader)? Also why did you choose to make CoroutineHelper an internal class and make no internal helper class for pause manager?

Thanks in advance!

Reply:
My rationale was based on "Does it need to be a game object".
PauseManager needs to receive and act on Update and FixedUpdate, for that reason it was really easy to make it a MonoBehavior and just say "you need this".
The main reason for the Coroutine system is to allow non-game objects to use coroutines (and easier lambda style things), CoroutineHelper is just an annoyance.
In summary I made it internal because I wanted CoroutineHelper to be a hidden implementation detail of the coroutine system, but I didn't worry about that abstraction for PauseManager. Whether or not I should have is a good question tbh.